ark/phenotyping/som_utils.py

```diff
@@ -13,6 +13,60 @@
 from ark.utils import misc_utils
 
 
+def compute_cluster_avg(fovs, channels, base_dir,
+                        cluster_dir='pixel_mat_clustered',
+                        cluster_avg_name='pixel_cluster_avg.feather'):
+    """Averages channel values across all fovs in pixel_mat_clustered
+
+    Args:
+        fovs (list):
+            The list of fovs to subset on
+        channels (list):
+            The list of channels to subset on
+        base_dir (str):
+            Name of the directory to save the pixel files to
+        cluster_dir (str):
+            Name of the file containing the pixel data with cluster labels
+        cluster_avg_name (str):
+            Name of file to save the averaged results to
+    """
+
+    cluster_avgs = None
+
+    for fov in fovs:
+        # read in the fovs data
+        fov_pixel_data = feather.read_dataframe(
+            os.path.join(base_dir, cluster_dir, fov + '.feather'))
+
+        # aggregate the sums and counts
+        sum_by_cluster = fov_pixel_data.groupby('cluster')[channels].sum()
+        count_by_cluster = fov_pixel_data.groupby('cluster')[channels].size().to_frame('count')
+
+        # concat the results together
+        agg_results = pd.merge(
+            sum_by_cluster, count_by_cluster, left_index=True, right_index=True).reset_index()
+
+        # set to cluster_avgs if not defined, otherwise concat
+        if cluster_avgs is None:
+            cluster_avgs = agg_results
+        else:
+            cluster_avgs = pd.concat([cluster_avgs, agg_results])
```

Comment:
I think if you do
```
cluster_avgs = pd.DataFrame()
```
as the initialization in line 34, you can always concatenate to it and won't have to do this check